server/app/views/admin/programs/ProgramBlockPredicateConfigureView.java

```diff
@@ -0,0 +1,621 @@
+
+                  if (maybeOperator.isPresent()
+                      && operator.name().equals(maybeOperator.get().name())) {
+                    optionTag.isSelected();
+                  }
+
+                  finalMaybeSelectedScalar.ifPresent(
+                      selectedScalar -> {
+                        if (!operator.getOperableTypes().contains(selectedScalar.toScalarType())) {
+                          optionTag.withClass("hidden");
+                        }
+                      });
+
+                  return optionTag;
+                })
+            .collect(toImmutableList());
+
+    return new SelectWithLabel()
+        .setFieldName(String.format("question-%d-operator", questionDefinition.getId()))
+        .setCustomOptions(operatorOptions)
+        .addReferenceClass(ReferenceClasses.PREDICATE_OPERATOR_SELECT)
+        .getSelectTag()
+        .withData("question-id", String.valueOf(questionDefinition.getId()));
+  }
+
+  private DivTag createValueField(
+      QuestionDefinition questionDefinition,
+      int groupId,
+      Optional<Scalar> maybeScalar,
+      Optional<PredicateValue> maybePredicateValue) {
+    DivTag valueField = div().withClasses(COLUMN_WIDTH);
+
+    if (questionDefinition.getQuestionType().isMultiOptionType()) {
+      // If it's a multi-option question, we need to provide a discrete list of possible values to
+      // choose from instead of a freeform text field. Not only is it a better UX, but we store the
+      // ID of the options rather than the display strings since the option display strings are
+      // localized.
+      ImmutableList<QuestionOption> options =
+          ((MultiOptionQuestionDefinition) questionDefinition).getOptions();
+
+      ImmutableSet<String> currentlyCheckedValues =
+          maybePredicateValue
+              .map(PredicateValue::value)
+              .map(
+                  value ->
+                      Splitter.on(", ")
+                          .splitToStream(value.substring(1, value.length() - 1))
+                          .map(item -> item.replaceAll("\"", ""))
+                          .collect(ImmutableSet.toImmutableSet()))
+              .orElse(ImmutableSet.of());
+
+      for (QuestionOption option : options) {
+        boolean isChecked = currentlyCheckedValues.contains(String.valueOf(option.id()));
+
+        LabelTag optionCheckbox =
+            FieldWithLabel.checkbox()
+                .setFieldName(
+                    String.format(
+                        "group-%d-question-%d-predicateValues[]",
+                        groupId, questionDefinition.getId()))
+                .setValue(String.valueOf(option.id()))
+                .setLabelText(option.optionText().getDefault())
+                .setChecked(isChecked)
+                .getCheckboxTag();
+        valueField.with(optionCheckbox);
+      }
+
+      return valueField.withData("question-id", String.valueOf(questionDefinition.getId()));
+    } else {
```

Comment:
Don't need this else because of the return.  unnesting aids readability.

Reply:
done